src/screens/Inventory/AddInventory.tsx

```diff
@@ -0,0 +1,99 @@
+import { createStyles, FormControl, TextField as MaterialTextField, Theme } from '@material-ui/core';
+import { withStyles } from '@material-ui/core/styles';
+import { Autocomplete } from '@material-ui/lab';
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import { RouteComponentProps, useHistory } from 'react-router-dom';
+import BaseScreen from '../../components/BaseComponents/BaseScreen';
+import Button from '../../components/Button';
+import TextField from '../../components/TextField';
+import { InventoryRecord, ProductRecord, SiteRecord } from '../../lib/airtable/interface';
+import { createInventory } from '../../lib/airtable/request';
+import { addInventoryToRedux } from '../../lib/redux/inventoryData';
+import { EMPTY_INVENTORY, SiteInventoryData } from '../../lib/redux/inventoryDataSlice';
+import { EMPTY_SITE } from '../../lib/redux/siteDataSlice';
+import { RootState } from '../../lib/redux/store';
+
+const styles = (theme: Theme) =>
+  createStyles({
+    content: {
+      color: theme.palette.text.primary,
+    },
+    formControl: {
+      width: '100%',
+    },
+  });
+
+interface AddInventoryProps extends RouteComponentProps {
+  classes: { content: string, formControl: string};
+  location: any;
+  currentSite: SiteRecord;
+  products: Record<string, ProductRecord>;
+  userId: string;
+  sitesInventory: Record<string, SiteInventoryData>;
+}
+
+
+function AddInventory(props: AddInventoryProps) {
+  const { classes, currentSite, products, userId, sitesInventory } = props;
+  const history = useHistory();
+  // Products that the site already has inventory for
+  const currentSiteProducts = sitesInventory[currentSite.id].siteInventory.map((inventory: InventoryRecord) => inventory.productId);
+  {/* Only show products that the site doesn't already have */}
+  const productOptions = Object.entries(products).filter(([id, _]) => !currentSiteProducts.includes(id)).map(item => item[0]);
+  const [selectedProductId, setSelectedProductId] = useState("");
+  const [startingAmount, setStartingAmount] = useState(0);
+
+  // TODO: Add form input validation and error messaging
+  const handleSubmit = (event: React.MouseEvent) => {
+    // Prevent page refresh on submit
+    event.preventDefault();
+
+    // Make a deep copy of an empty inventory record
+    const inventory = JSON.parse(JSON.stringify(EMPTY_INVENTORY));
+    inventory.productId = selectedProductId;
+    inventory.siteId = currentSite.id;
+    inventory.currentQuantity = startingAmount;
+    inventory.periodStartQuantity = startingAmount;
+    addInventoryToRedux(inventory);
```

Comment:
The inventory's id should be added if the user is online

![image](https://user-images.githubusercontent.com/35501399/111924393-88695f80-8a61-11eb-9432-51eb41e49789.png)

This can probably be done as a promise to `createInventory`, but be aware of the race condition between persisting the `inventory` object to redux and the API request finishing (since these 2 operations are happening in parallel right now). Might be good to just primitively `await` the API call in a try statement, grab the id (and catch if offline), and `addInventoryToRedux(inventory)` after placing the id in `inventory`